### PR TITLE
[tinyfiledialogs] download from archive, not temporary ref

### PR DESCRIPTION
**Pull request by MaxKruse**
**Describe the pull request**
build the tinyfiledialogs lib consistently.

- What does your PR fix? 
Fixes issue #9665 (old issue, still persistent though)
- Which triplets are supported/not supported? Have you updated the CI baseline?
static-only lib, all tripplets

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
I have additionally messaged the original maintainer about a fixed-version download, so we will finally have some consistent building of this package.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @MaxKruse 
Thanks for this PR.
I noticed these failures on current CI pipeline:
```
##[error] PASSING, REMOVE FROM FAIL LIST: tinyfiledialogs:x64-linux (/ci/myagent/_work/3/s/scripts/ci.baseline.txt)
##[error] PASSING, REMOVE FROM FAIL LIST: tinyfiledialogs:x64-osx (/Volumes/work/1/s/scripts/ci.baseline.txt)
##[error] PASSING, REMOVE FROM FAIL LIST: tinyfiledialogs:x86-windows (C:\vsts\_work\4\s\scripts\ci.baseline.txt)
##[error] PASSING, REMOVE FROM FAIL LIST: tinyfiledialogs:x64-windows (C:\vsts\_work\4\s\scripts\ci.baseline.txt)
##[error] PASSING, REMOVE FROM FAIL LIST: tinyfiledialogs:x64-windows-static (C:\vsts\_work\4\s\scripts\ci.baseline.txt)
##[error] PASSING, REMOVE FROM FAIL LIST: tinyfiledialogs:arm64-windows (C:\vsts\_work\4\s\scripts\ci.baseline.txt)
```
Since this port has passed on these triplets, we should remove these codes from `ci.baseline.txt.`
Could you please update `ci.baseline.txt` and also bump the version in `CONTROL `file?

Please also help check if this port can support `uwp`. If not, please add `vcpkg_fail_port_install(ON_TARGET "uwp")` to the top of `portfile.cmake` and add `Supports: !uwp` to `CONTROL `file. If yes, please try to fix the regressions.

**Comment by MaxKruse** [comment]
Hi,

a change in version is not required, the library still has the same version as is.
ci.baseline.txt should also be correct now.

The library was not intended for UWP, so no support is given. I would assume its best to just keep it as an expected fail in that case.

Thanks for the help so far by the way! Have a great weekend.

**Comment by NancyLi1013** [comment]
Hi @MaxKruse 

Thanks for your update and above info.
It seems that CI tests are good except osx pipeline is running now.

Also, the version in CONTROL file needs to be updated as `3.4.3-1` . Since you have updated portfile.cmake.
This is not related to the library version itself. If there are some changes in portfile.cmake, the version is required to update.
BTW, could you please add `Supports: !uwp` to the bottom of CONTROL file? 
It seems a bit strange now.

**Comment by MaxKruse** [comment]
Sure thing. Weird how only OSX is stuck :/ its supposed to work fine on anything past 10.4

**Comment by MaxKruse** [comment]
@NancyLi1013 It seems that the OSX Test Server has hang-up about 8 hours ago. Pretty sure somebody should check that out!

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Thanks for your attention.
osx has passed now.

LGTM and thanks for your PR.

**Comment by MaxKruse** [comment]
quick reminder that this hasnt been merged yet 😗

File: ports/tinyfiledialogs/portfile.cmake [hunk 1]
```diff
@@ -1,9 +1,17 @@
+vcpkg_fail_port_install(ON_TARGET "uwp")
+
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
-vcpkg_from_git(
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://sourceforge.net/projects/tinyfiledialogs/files/v3.4/tinyfiledialogs-3.4.3.zip/download"
+    FILENAME "tinyfiledialogs.zip"
```
[the remark this answers is not in the document]
Comment (MaxKruse) [reply ports/tinyfiledialogs/portfile.cmake]: Roger Roger

**Comment by strega-nil** [comment]
Alright, thanks much @MaxKruse !!!

**Pull request merged**